pydash/collections.py

```diff
@@ -1049,31 +1049,51 @@ def sort_by(collection, iteratee=None, reverse=False):
     return sorted(collection, key=pyd.iteratee(iteratee), reverse=reverse)
 
 
-def to_list(collection):
-    """Converts the collection to a list.
+def to_list(item):
+    """ Converts a collection, an iterable or a single item to a list.
 
     Args:
-        collection (list|dict): Collection to iterate over.
+        item (mixed): Item to convert or wrap.
 
     Returns:
-        list: Collection converted to list.
+        list: Converted or wrapped item..
 
     Example:
 
         >>> results = to_list({'a': 1, 'b': 2, 'c': 3})
         >>> assert set(results) == set([1, 2, 3])
+
         >>> to_list((1, 2, 3, 4))
         [1, 2, 3, 4]
 
+        >>> to_list(1)
+        [1]
+
+        >>> to_list([1])
+        [1]
+
+        >>> to_list(a for a in [1, 2, 3])
+        [1, 2, 3]
+
     .. versionadded:: 1.0.0
+
+    .. versionchanged:: 4.3.0
+        Wrap non iterables items in a list.
+        Convert other iterables to list.
+
     """
-    if isinstance(collection, dict):
-        ret = collection.values()
-    else:
-        ret = list(collection)
 
-    return ret
+    if isinstance(item, list):
+        return item
 
+    elif isinstance(item, dict):
+        return item.values()
+
+    elif hasattr(item, '__iter__'):
+        return list(item)
+
+    else:
+        return [item]
```

Comment:
The reason this is failing on Travis is because Python 2.7 `str` doesn't have an `__iter__` method. So with these changes, on Python 3, `to_list('abc') == ['a', 'b', 'c']` but on Python 2 it's `to_list('abc') == ['abc']`.

The current behavior is that `to_list('abc')` should be `['a', 'b', 'c']`. Therefore, this could be simplified to:

```python
if isinstance(collection, dict):
    ret = collection.values()
else:
    try:
        ret = list(collection)
    except Exception:
        ret = [collection]

return ret
```

This next question then is whether you are proposing that `to_list('abc')` should return `['abc']`? Changing this behavior would break backwards compatibility so cannot do that on the `v4` line.

Reply:
Yes i saw it. Unfortunately my local test environment offers only python3.6, what somehow works not so well with tox for me. Anyway. 

I'm proposing that converting a single string to a list shot wrap this string into a list and not split it by character. We could add an parameter for this, but actually the python2 behavior feels more natural for me.

Reply:
In that case, adding a parameter for making strings return as a wrapped list would be fine. Next major release could make that the default behavior potentially.